generate/generate.go

```diff
@@ -81,6 +87,34 @@ func Do(rootdir string, workingDir string) Report {
 		var generated []fileInfo
 		report := dirReport{}
 
+		logger.Trace().Msg("loading asserts for stack")
+
+		asserts, err := loadAsserts(projmeta, stack, globals)
+		if err != nil {
+			report.err = err
+			return report
+		}
+
+		logger.Trace().Msg("checking stack asserts")
+		errs := errors.L()
+		for _, assert := range asserts {
+			if !assert.Assertion {
+				assertRange := assert.Range
```

Comment:
maybe we could have our own Range type where the filename is always a project.Path, and then a function to do conversion (we do this on other places too I think, or something similar at least). not sure, just an idea.

Reply:
👍